HHS/Head-Start-TTADP
# [TTAHUB-502] Add AWS Elasticsearch text search filter

## Description of change

**NOTE:** This contains required changes from https://github.com/HHS/Head-Start-TTADP/pull/1250 also under review.

This add a 'Text' filter to the AR landing to allow the search of the 'activityreport' index documents.

![image](https://user-images.githubusercontent.com/84350609/211858618-e6df14b6-aaed-4f94-96f4-2247d96ce118.png)

Keep in mind that this searches reports that have been indexed via a submit or the indexing job (approved reports).

It currently searches the following fields:

- context
- recipient next steps
- specialist next steps
- goal text
- objective text
- objective tta provided

It may also return results that contain **partial matches**.

## How to test

_**Prerequisites:**_ Before testing you should run the index script '**createAwsElasticSearchIndexes**'. Once completed you will need to make sure both **DOCKER** and the **WORKER** are running.

To start I would recommend creating and submitting a report that has unique values for each of the above fields. Then create 'Text' filter on the AR landing and make sure you get the desired results. Also test this same filter in conjunction with other filters.

Finally test the new filter against the existing reports. 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-502


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
